atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -132,12 +164,27 @@ private boolean overflowColumnExists(String shortTableName) {
                         shortTableName.toUpperCase());
     }
 
+    private boolean tableExists() {
+        return conns.get()
+                .selectExistsUnregisteredQuery(
+                        "SELECT 1 FROM " + config.metadataTable().getQualifiedName() + " WHERE table_name = ?",
+                        tableRef.getQualifiedName());
+    }
+
+    private boolean overflowTableExists() {
+        try {
+            String shortTableName = oracleTableNameGetter.getInternalShortOverflowTableName(conns, tableRef);
+            return conns.get()
+                    .selectExistsUnregisteredQuery(
+                            "SELECT 1 FROM user_tables WHERE TABLE_NAME = ?", shortTableName.toUpperCase());
+        } catch (TableMappingNotFoundException e) {
+            log.warn("Table mapping does not exist but is expected for overflow table.", e);
```

Comment:
Isn't this a bad state? (i.e., we expect there to be an overflow table or we should throwForMissingOverflowTable earlier at the value style part)?

Reply:
(There's a separate thing to note here too: We now throw when the overflow table is missing but we expect it to exist (as we should have thrown at throwForMissingOverflowTable) if table mapping is enabled, but we don't do the same for unmapped tables. I don't think it matters terribly so, but just pointing it out)